test/framework/src/main/java/org/elasticsearch/test/hamcrest/ElasticsearchAssertions.java

```diff
@@ -364,6 +369,41 @@ public static void assertResponse(ActionRequestBuilder<?, SearchResponse> search
         }
     }
 
+    public record ScrollResponses(
+        SearchResponse initialSearchResponse,
+        List<SearchResponse> scrollResponses,
+        List<SearchResponse> allResponses
+    ) {}
+
+    public static void assertScrollResponses(
+        int timoutSeconds,
+        SearchRequestBuilder searchRequestBuilder,
+        Consumer<ScrollResponses> consumer
+    ) {
+        searchRequestBuilder.setScroll(TimeValue.timeValueSeconds(timoutSeconds));
+        List<SearchResponse> responses = new ArrayList<>();
+        var scrollResponse = searchRequestBuilder.get();
+        responses.add(scrollResponse);
+        try {
+            while (scrollResponse.getHits().getHits().length > 0) {
+                scrollResponse = client().prepareSearchScroll(scrollResponse.getScrollId())
```

Comment:
Lets add to ESIngestTestCase a method so we don't have to create the query here. Something like:

```
 public static SearchScrollRequestBuilder prepareScrollSearch(String scrollId, TimeValue timeout) {
        return client().prepareSearchScroll(scrollId).setScroll(timeout);
   }
```